### ISSUE TITLE
Learning rate finder

### ISSUE THREAD

**Issue by aymuos15** [aymuos15/AutoMONAI#32]
## Motivation

Choosing a good learning rate is one of the most impactful hyperparameter decisions. A LR finder (Leslie Smith's range test) sweeps across LR values in one short run and identifies the optimal range — saving hours of manual tuning.

### CLI
```
--lr_find               Run LR finder before training and use the result (default: false)
--lr_find_start         Start of LR range (default: 1e-7)
--lr_find_end           End of LR range (default: 1.0)
--lr_find_steps         Number of steps in the sweep (default: 100)
```

### How it works
1. Before the training loop, run a short sweep: increase LR exponentially from `lr_find_start` to `lr_find_end` over `lr_find_steps` mini-batches
2. Record loss at each step
3. Pick the LR where loss is decreasing fastest (steepest negative gradient)
4. Set that as the LR for actual training
5. Reset model weights to initial state

### Implementation

**`automonai/core/lr_finder.py`**:
```python
def find_lr(fabric, model, train_loader, loss_fn, optimizer, start_lr, end_lr, steps):
    """Run LR range test. Returns suggested_lr and (lrs, losses) for plotting."""
```

**`run.py` changes**:
- When `--lr_find` is set, run `find_lr()` after model setup but before the training loop
- Print: `LR Finder: suggested lr = 3.2e-4`
- Save plot to run directory: `lr_finder.png`
- Override `--lr` with the found value

### W&B logging
- Log `lr_finder/lr` vs `lr_finder/loss` as a W&B table/chart
- `lr_finder/suggested_lr` in run config

### UI
- **Find LR** checkbox in Generate tab next to the LR input
- After finding: auto-populate the LR field with the suggested value

### Dependencies
None — just a training loop variant with exponential LR schedule. No external library needed.
